File: quickstart.mjs
/**
 * QuanticData API quickstart — Node.js 18+ (native fetch, zero dependencies).
 *   QUANTICDATA_API_KEY=qd_live_... node quickstart.mjs
 */
const BASE = "https://api.quanticdata.io/v1";
const KEY = process.env.QUANTICDATA_API_KEY;
if (!KEY) {
  console.error("set QUANTICDATA_API_KEY (get one at https://app.quanticdata.io/register)");
  process.exit(1);
}

/** POST to the API and unwrap the { type, message, payload } envelope. */
async function call(path, body) {
  const res = await fetch(BASE + path, {
    method: "POST",
    headers: { Authorization: `Bearer ${KEY}`, "Content-Type": "application/json" },
    body: JSON.stringify(body),
  });
  const data = await res.json();
  if (!res.ok || data.type === "error") {
    throw new Error(`${path} failed (${res.status}): ${data.message}`);
  }
  return data.payload ?? {};
}

const page = await call("/scrape", { url: "https://example.com", format: "markdown" });
console.log("--- markdown -------------------------------------------------");
console.log((page.markdown ?? "").slice(0, 400));
console.log("title:", page.metadata?.title);
console.log("cost :", page.usage?.cost_usd);

const serp = await call("/serp", { query: "web scraping api", country: "us", num: 5 });
console.log("\n--- organic --------------------------------------------------");
for (const row of (serp.organic ?? []).slice(0, 5)) {
  console.log(`${String(row.position).padStart(2)}. ${row.title}\n    ${row.link}`);
}

File: quickstart.rb
# QuanticData API quickstart — Ruby (net/http, stdlib only).
#   QUANTICDATA_API_KEY=qd_live_... ruby quickstart.rb
require "json"
require "net/http"
require "uri"

BASE = "https://api.quanticdata.io/v1".freeze

KEY = ENV["QUANTICDATA_API_KEY"]
abort "set QUANTICDATA_API_KEY — https://app.quanticdata.io/register" if KEY.nil? || KEY.empty?

# POST to the API and unwrap the { type, message, payload } envelope.
def qd_call(path, body)
  uri = URI("#{BASE}#{path}")
  http = Net::HTTP.new(uri.host, uri.port)
  http.use_ssl = true
  http.read_timeout = 120

  req = Net::HTTP::Post.new(uri)
  req["Authorization"] = "Bearer #{KEY}"
  req["Content-Type"] = "application/json"
  req.body = JSON.dump(body)

  res = http.request(req)
  data = JSON.parse(res.body) rescue {}
  if res.code.to_i >= 400 || data["type"] == "error"
    raise "#{path} failed (#{res.code}): #{data['message']}"
  end
  data["payload"] || {}
end

page = qd_call("/scrape", { url: "https://example.com", format: "markdown" })
puts "--- markdown -------------------------------------------------"
puts page["markdown"].to_s[0, 400]
puts "title: #{page.dig('metadata', 'title')}"
puts "cost : #{page.dig('usage', 'cost_usd')}"

serp = qd_call("/serp", { query: "web scraping api", country: "us", num: 5 })
puts
puts "--- organic --------------------------------------------------"
serp.fetch("organic", []).first(5).each do |row|
  puts format("%2d. %s\n    %s", row["position"].to_i, row["title"], row["link"])
end
